test/llvm/lifetime-intrinsics/lifetime-multi-return.chpl

```diff
@@ -0,0 +1,66 @@
+// This test is used to verify that llvm.lifetime instrinsics are being emitted
+// for procedures having multiple return statements (gotoReturns)
+
+// CHECK: %[[REG1:[0-9]+]] = bitcast double* %ret_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.start.p0i8(i64 8, i8* %[[REG1]])
+// CHECK: %[[REG2:[0-9]+]] = bitcast double* %y_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.start.p0i8(i64 8, i8* %[[REG2]])
+// CHECK: %[[REG3:[0-9]+]] = bitcast double* %z_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.start.p0i8(i64 8, i8* %[[REG3]])
+// CHECK: %[[REG4:[0-9]+]] = bitcast double** %i_lhs_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.start.p0i8(i64 8, i8* %[[REG4]])
+// CHECK: %[[REG5:[0-9]+]] = bitcast double** %i_lhs_chpl2 to i8*
+// CHECK-NEXT: call void @llvm.lifetime.start.p0i8(i64 8, i8* %[[REG5]])
+// CHECK: %[[REG6:[0-9]+]] = bitcast double* %coerce_tmp_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.start.p0i8(i64 8, i8* %[[REG6]])
+
+// CHECK: %[[REG7:[0-9]+]] = bitcast double* %y_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG7]])
+// CHECK: %[[REG8:[0-9]+]] = bitcast double* %z_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG8]])
+// CHECK: %[[REG9:[0-9]+]] = bitcast double* %coerce_tmp_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG9]])
+// CHECK: %[[REG10:[0-9]+]] = bitcast double** %i_lhs_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG10]])
+// CHECK: %[[REG11:[0-9]+]] = bitcast double** %i_lhs_chpl2 to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG11]])
+// CHECK: br label %_end{{.*}}
+
+// CHECK: %[[REG12:[0-9]+]] = bitcast double* %coerce_tmp_chpl2 to i8*
+// CHECK-NEXT: call void @llvm.lifetime.start.p0i8(i64 8, i8* %[[REG12]])
+
+// CHECK: %[[REG13:[0-9]+]] = bitcast double* %y_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG13]])
+// CHECK: %[[REG14:[0-9]+]] = bitcast double* %z_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG14]])
+// CHECK: %[[REG15:[0-9]+]] = bitcast double* %coerce_tmp_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG15]])
+// CHECK: %[[REG16:[0-9]+]] = bitcast double** %i_lhs_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG16]])
+// CHECK: %[[REG17:[0-9]+]] = bitcast double** %i_lhs_chpl2 to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG17]])
+// CHECK: %[[REG18:[0-9]+]] = bitcast double* %coerce_tmp_chpl2 to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG18]])
+// CHECK: br label %_end{{.*}}
+
+// CHECK: %[[REG19:[0-9]+]] = bitcast double* %ret_chpl to i8*
+// CHECK-NEXT: call void @llvm.lifetime.end.p0i8(i64 8, i8* %[[REG19]])
```

Comment:
I'm expecting to see checks for lifetime ends for x and a at the end of the function.

Additionally, you can interleave these CHECK comments with the body of the function. I think that would make the situation clearer in this case.

Reply:
`x` and `a` are constants so should they still have a lifetime?

Reply:
Yes, they are still allocated on the stack. (a `param` would be a compile-time constant, a `const` in Chapel can vary at runtime, it just keeps its value once initialized).

Reply:
I suppose that since `x` and `a` both do not have a `lifetime.start` attached to them since they never change once assigned, they won't have a `lifetime.end` either.